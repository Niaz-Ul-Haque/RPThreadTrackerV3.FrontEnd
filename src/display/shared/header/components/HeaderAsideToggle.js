// #region imports
import React from 'react';
import PropTypes from 'prop-types';
import { NavItem, NavLink, Badge } from 'reactstrap';
// #endregion imports

const propTypes = {
	asideToggle: PropTypes.func.isRequired,
	newsUnreadCount: PropTypes.number.isRequired
};

const HeaderAsideToggle = (props) => {
	const {
		asideToggle,
		newsUnreadCount
	} = props;

	return (
		<NavItem>
			<NavLink
				data-spec="header-aside-toggle-link"
				href="#"
				onClick={asideToggle}
			>
				<i className="icon-bell" />
				{
					newsUnreadCount > 0 &&
					<Badge data-spec="header-aside-toggle-unread-badge" pill color="danger">
						{newsUnreadCount}
					</Badge>
				}
			</NavLink>
		</NavItem>
	);
};
HeaderAsideToggle.propTypes = propTypes;
export default HeaderAsideToggle;
