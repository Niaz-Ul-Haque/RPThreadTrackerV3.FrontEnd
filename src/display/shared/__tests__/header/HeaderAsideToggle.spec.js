// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '../../../../../config/tests/helpers.unit';
import HeaderAsideToggle from '../../header/components/HeaderAsideToggle';
// #endregion imports

const createTestProps = propOverrides => ({
	asideToggle: jest.fn(),
	newsUnreadCount: 0,
	isNewsAsideOpen: true,
	...propOverrides
});

describe('rendering', () => {
	it('should render valid snapshot with no new items', () => {
		const props = createTestProps();
		const jsx = (<HeaderAsideToggle {...props} />);
		const element = shallow(jsx);
		expect(element).toMatchSnapshot();
	});
	it('should render valid snapshot with new items', () => {
		const props = createTestProps({ newsUnreadCount: 3 });
		const jsx = (<HeaderAsideToggle {...props} />);
		const element = shallow(jsx);
		expect(element).toMatchSnapshot();
	});
	it('should not render badge when no new items', () => {
		const props = createTestProps();
		const jsx = (<HeaderAsideToggle {...props} />);
		const element = shallow(jsx);
		expect(getSpecWrapper(element, 'header-aside-toggle-unread-badge')).toHaveLength(0);
	});
	it('should render badge when new items', () => {
		const props = createTestProps({ newsUnreadCount: 3 });
		const jsx = (<HeaderAsideToggle {...props} />);
		const element = shallow(jsx);
		const badge = getSpecWrapper(element, 'header-aside-toggle-unread-badge').children().text();
		expect(badge).toBe('3');
	});
});

describe('behavior', () => {
	it('should send toggled aside state to handler on click', () => {
		const asideToggle = jest.fn();
		const props = createTestProps({ asideToggle });
		const jsx = (<HeaderAsideToggle {...props} />);
		const element = shallow(jsx);
		const link = getSpecWrapper(element, 'header-aside-toggle-link');
		link.simulate('click');
		expect(asideToggle).toHaveBeenCalledTimes(1);
		expect(asideToggle).toHaveBeenCalledWith(false);
	});
});
