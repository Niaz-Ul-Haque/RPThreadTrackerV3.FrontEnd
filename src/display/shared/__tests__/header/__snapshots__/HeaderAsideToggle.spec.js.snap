// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render valid snapshot with new items 1`] = `
<NavItem
  tag="li"
>
  <NavLink
    data-spec="header-aside-toggle-link"
    href="#"
    onClick={[MockFunction]}
    tag="a"
  >
    <i
      className="icon-bell"
    />
    <Badge
      color="danger"
      data-spec="header-aside-toggle-unread-badge"
      pill={true}
      tag="span"
    >
      3
    </Badge>
  </NavLink>
</NavItem>
`;

exports[`rendering should render valid snapshot with no new items 1`] = `
<NavItem
  tag="li"
>
  <NavLink
    data-spec="header-aside-toggle-link"
    href="#"
    onClick={[MockFunction]}
    tag="a"
  >
    <i
      className="icon-bell"
    />
  </NavLink>
</NavItem>
`;
